File: c/main3.c
#include <stdio.h>

void main(int argc, char** argv)
{
    if( argc > 1 ) {
    	printf("String : %s\n", argv[ 1 ] );
    	printf("Integer: %i\n", get_integer_from_string( argv[ 1 ] ));
    }
}

/**
 * Function parses String to Int. Parsing stops if value outside 0 - 9 is met.
 * '-' char is accepted as indicator of a negative number.
 *
 * Param  String to be parsed.
 * Return Parsed Integer.
 */
int get_integer_from_string( const char* str ) {
	int value = 0, num = 0, sign = 1;
	const char* c = NULL;

	// Is the number negative
	if( str[0] == '-') {
		sign = -1;
		str++;
	}

	// Parse int
	for(c = str; *c != '\0'; c++){
		num = *c - '0';
		if( num < 0 || num > 9 ) break;
		value = value * 10 + num;
	}

	return value * sign;
}

File: nodejs/main3.js
(function () {
	"use strict";

	console.log( 'String  s: ', process.argv[2] );
	console.log( 'Integer i: ', get_integer_from_string( process.argv[2] ));

	function get_integer_from_string( str ) {
		var num = 1, value = 0, sign = 1, index = 0;
		var charCodeOfZero = '0'.charCodeAt(0);

		// Is the number negative or positive
		if( str[0] == '-') {
			sign = -1;
			index = 1;
		}

		// Parse int
		for( index; index < str.length; index++ ) {
			num = str.charCodeAt( index ) - charCodeOfZero;
			if( num < 0 || num > 9 ) break;
			value = value * 10 + num;
		}

		return value * sign;
	}
})();
